lib/core/AxiosHeaders.js

```diff
@@ -89,7 +89,11 @@ class AxiosHeaders {
       const key = utils.findKey(self, lHeader);
 
       if(!key || self[key] === undefined || _rewrite === true || (_rewrite === undefined && self[key] !== false)) {
-        self[key || _header] = normalizeValue(_value);
+        if (key && self[key] != undefined && self[key] !== false && _rewrite === false) {
+          self[key] = (Array.isArray(self[key]) ? self[key] : [self[key]]).concat(normalizeValue(_value));
```

Comment:
```suggestion
          const arrayToConcat = Array.isArray(self[key]) ? self[key] : [self[key]];
          self[key] = (arrayToConcat).concat(normalizeValue(_value));
```

Maybe it's better to name `arrayToConcat` constant in other way but I guess it's better to have separate constant for that.

Reply:
have renamed to `headerValues`